doc/dev/kclient_dev_bootstrap.rst

```diff
@@ -0,0 +1,474 @@
+
+
+Finishing Up
+^^^^^^^^^^^^
+
+To access the VM display, connect to ``localhost:5901`` with a VNC client, where
+you will be able to interact with the VM’s graphical installer.
+
+After the VM boots, proceed with the OS installation process as prompted. After
+the OS is fully installed, you can remove the ISO reference from future
+commands in order to boot directly from the virtual disk.
+
+SSH Port Forwarding for Remote VNC Access
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you need to access the VM’s VNC display from a remote machine, you can set
+up SSH port forwarding to securely tunnel the VNC connection over SSH. This
+allows you to access VNC as if it were on your local network, even when
+connecting remotely.
+
+On your **local machine** (from which you want to access the remote VM), run
+the following SSH command:
+
+.. prompt:: bash #
+
+   ssh -L 5901:localhost:5901 user@remote_host
+
+In this command:
+
+- ``-L 5901:localhost:5901`` sets up local port forwarding, mapping your local
+  port ``5901`` to ``localhost:5901`` on the remote host.
+- ``user@remote_host`` is the SSH username and address of the remote host where
+  the VM is running.
+
+Once connected, open your VNC client on the local machine and connect to
+``localhost:5901``. This setup will forward the VNC traffic securely through
+SSH, allowing you to interact with the VM from a remote location.
+
+Differential Copies with QCOW2 and qemu-img
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The QCOW2 (QEMU Copy On Write) disk format in QEMU is highly versatile, as it
+supports snapshotting and differential copies. With ``qemu-img``, you can
+create a differential copy (or "backing file") that references a base image
+without duplicating the entire disk. This approach allows for efficient storage
+and versioning of virtual machine states.
+
+To create a differential image, use ``qemu-img`` to specify a new QCOW2 image
+with a backing file. For example:
+
+.. prompt:: bash #
+
+  qemu-img create -f qcow2 -b base_image.qcow2 differential_image.qcow2
+
+Here:
+
+- ``-f qcow2`` specifies the QCOW2 format.
+- ``-b base_image.qcow2`` sets ``base_image.qcow2`` as the backing file, which
+  holds the base disk state.
+- ``differential_image.qcow2`` will store only changes made since the base
+  image, saving disk space.
+
+When the VM runs with ``differential_image.qcow2``, QEMU reads data from the
+base file unless there are modifications in the differential image. This allows
+you to maintain a series of snapshots, with each differential image capturing
+incremental changes, useful for testing, development iterations, or backup
+purposes.
+
+Setting Up a 9P Shared Folder Between Host and Guest
+-----------------------------------------------------
+
+The 9P (Plan 9 Filesystem Protocol) allows you to share directories between the
+host and guest in QEMU, making file transfers between the two environments
+easy. Follow these steps to set up a 9P shared folder.
```

Comment:
Does this have some utility you actually use in this guide or just a convenient file share you sometimes use?

If the latter, let's move it out of the base doc and link to it in the new "Advanced debugging" section in `kclient.rst`.

Reply:
@batrick I'll answer in one place.

1. Fast compile - localmodconfig -- removes all redundant kernel modules and this is why you are using the guest config file, you don't want to be missing something you need for booting -- its an extreme case but also goot practice.
2. Fast Iterations - The code compiles on the host where you have your IDE,Caches, etc..., but that happens is the shared folder so the binaries are visible in the guest -- so P9 or other shared folders are a necessity. 

Slack me if you think some things can be made clearer.

Reply:
@Markuze, Are you willing to let me make these changes?

Reply:
Sure, I would be grateful

Reply:
I have moved this into a file called 9p.rst, and I have linked to it where Patrick indicated it should be linked to.